### DISCUSSION TITLE
using filament in c++ with qt for andorid

### DISCUSSION THREAD

**Discussion by yygbruce** [google/filament#6648] [Q&A]
I use the libfilament-jni.so(from filament-android-release.aar) library with qt, and the app can compile successfully,  but when I run the app, it crash.I found that these codes cause the app to crash:
if (renderer_->beginFrame(swap_chain_)) {
    renderer_->render(view_);
    renderer_->endFrame();
    engine_->flushAndWait();
  }
when I removed these code, It can run successfully on the android phone, and the error message:
W Activity: PerfMonitor: Slow Operation: Activity org.qtproject.example/org.qtproject.qt5.android.bindings.QtActivity onCreate took 682ms
W Looper  : PerfMonitor looperActivity : package=org.qtproject.example/org.qtproject.qt5.android.bindings.QtActivity time=1ms latency=882ms running=0ms  procState=-1  historyMsgCount=4 (msgIndex=1 wall=801ms seq=4 h=android.app.ActivityThread$H w=159) (msgIndex=2 wall=801ms seq=4 h=android.app.ActivityThread$H w=159) (msgIndex=3 wall=84ms seq=3 h=android.app.ActivityThread$H w=110) (msgIndex=4 wall=801ms seq=4 h=android.app.ActivityThread$H w=159)
I AdrenoGLES-0: QUALCOMM build                   : 9b824c0bed, I6c7bff0fbd
I AdrenoGLES-0: Build Date                       : 10/09/21
I AdrenoGLES-0: OpenGL ES Shader Compiler Version: EV031.32.02.15
I AdrenoGLES-0: Local Branch                     :
I AdrenoGLES-0: Remote Branch                    :
I AdrenoGLES-0: Remote Branch                    :
I AdrenoGLES-0: Reconstruct Branch               :
I AdrenoGLES-0: Build Config                     : S P 10.0.7 AArch32
I AdrenoGLES-0: Driver Path                      : /vendor/lib/egl/libGLESv2_adreno.so
I AdrenoGLES-0: PFP: 0x016ee190, ME: 0x00000000
D hw-ProcessState: Binder ioctl to enable oneway spam detection failed: Invalid argument
D DecorView[]: onWindowFocusChanged hasWindowFocus true
I Filament: FEngine (32 bits) created at 0xedf79600 (threading is enabled)
D Filament: Using ASurfaceTexture
I Filament: FEngine resolved backend: OpenGL
V Filament: [Qualcomm], [Adreno (TM) 618], [OpenGL ES 3.2 V@0502.0 (GIT@9b824c0bed, I6c7bff0fbd, 1633761551) (Date:10/09/21)], [OpenGL ES GLSL ES 3.20]
V Filament: Feature level: 2
V Filament: Active workarounds:
V Filament: dont_use_timer_query
V Filament: disable_sidecar_blit_into_texture_array
V Filament: split_easu
V Filament: invalidate_end_only_if_invalidate_start
V Filament: allow_read_only_ancillary_feedback_loop
V Filament: enable_initialize_non_used_uniform_array
I Filament: FEngine feature level: 2
F libc    : Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x55 in tid 14317 (FEngine::loop), pid 14156 (project.example)
Has anyone encountered such a problem?

**Comment by romainguy** [comment]
How did you setup the swap chain?

**Comment by yygbruce** [comment]
Ok, I upgraded the qt's verson from 5.12.12 to 5.15.2, and it can work.Hope this is useful for somebody.
![Screenshot_2023-04-22-21-15-25-494_org qtproject example filament](https://user-images.githubusercontent.com/18020338/233788338-5e1b581d-1a82-4730-a799-dde899e7056d.jpg)
